### PR TITLE
Fix typos in Ristretto explicit decoding formulas

### PR DESCRIPTION
**Pull request by hdevalence**
For some reason these were different than both the derivation and also the test vectors (which were correct).

### CODE REVIEW

File: site/formulas/decoding.md [hunk 1]
```diff
@@ -7,14 +7,14 @@ follows.
 1. Check that `s_bytes` is the canonical encoding of a field element, or else abort.
 2. Decode `s_bytes`, a byte-string, into \\(s\\), a field element.
 3. Check that \\(s\\) is non-negative, or else abort.
-4. \\(u_1 \gets 1 - a s^{2} \\).
-5. \\(u_2 \gets 1 + a s^{2} \\).
+4. \\(u_1 \gets 1 + a s^{2} \\).
```
Comment (isislovecruft) [inline site/formulas/decoding.md:10]: I see that `u_1` is `1-as²` [here](https://github.com/dalek-cryptography/curve25519-dalek/blob/39145da396226516d8d1f663ad26f86bf04856db/src/ristretto.rs#L261). Is the code wrong too?

Comment (isislovecruft) [inline site/formulas/decoding.md:10]: Also, should we specify that `a = -1` somewhere?

File: site/formulas/decoding.md [hunk 2]
```diff
@@ -7,14 +7,14 @@ follows.
...
+5. \\(u_2 \gets 1 - a s^{2} \\).
```
Comment (isislovecruft) [inline site/formulas/decoding.md:11]: Similar as to above, I see that `u_2` is `1+as²` [in the code](https://github.com/dalek-cryptography/curve25519-dalek/blob/39145da396226516d8d1f663ad26f86bf04856db/src/ristretto.rs#L262).

File: site/formulas/decoding.md [hunk 3]
```diff
@@ -7,14 +7,14 @@ follows.
...
 6. \\(v \gets adu_1^2 - u_2^2 \\).
```
Comment (isislovecruft) [inline site/formulas/decoding.md:12]: I think `u_1` and `u_2` are actually switched here. The [code](https://github.com/dalek-cryptography/curve25519-dalek/blob/39145da396226516d8d1f663ad26f86bf04856db/src/ristretto.rs#L266) is

```
        // v == ad(1-as²)² - (1+as²)²            where d=-121665/121666
        let v = &(&(-&constants::EDWARDS_D) * &u2.square()) - &u1_sqr;
```
(And the comment in the code is also wrong.)

File: site/formulas/decoding.md [hunk 4]
```diff
@@ -7,14 +7,14 @@ follows.
...
-7. \\(I \gets \mathrm{invsqrt}( v u_1^{2} ) \\), or abort if the square root does not exist.
+7. \\(I \gets \mathrm{invsqrt}( v u_2^{2} ) \\), or abort if the square root does not exist.
```
Comment (isislovecruft) [inline site/formulas/decoding.md:13]: I see `I = 1/sqrt(v u_1)` in the [code](https://github.com/dalek-cryptography/curve25519-dalek/blob/39145da396226516d8d1f663ad26f86bf04856db/src/ristretto.rs#L268)?

File: site/formulas/decoding.md [hunk 5]
```diff
@@ -7,14 +7,14 @@ follows.
...
 8. \\(D_x \gets I u_2 \\).
```
Comment (isislovecruft) [inline site/formulas/decoding.md:14]: This should be `u_1` instead of `u_2` here.

File: site/formulas/decoding.md [hunk 6]
```diff
@@ -7,14 +7,14 @@ follows.
...
 9. \\(D_y \gets I D_x v \\).
 10. \\(x \gets |2 s D_x| \\), i.e., compute \\(2sD_x\\) and negate it if it is negative.
-11. Compute \\(y \gets u_2 D_y \\).
+11. Compute \\(y \gets u_1 D_y \\).
```
Comment (isislovecruft) [inline site/formulas/decoding.md:17]: This should also be `u_2` here. The [code](https://github.com/dalek-cryptography/curve25519-dalek/blob/39145da396226516d8d1f663ad26f86bf04856db/src/ristretto.rs#L279) is:

```
        // y == (1-as²)/(1+as²)
        let y = &u2 * &Dy;
```

(And again the comment in the code is wrong.)

**Review by isislovecruft — CHANGES_REQUESTED** [review]

File: site/formulas/decoding.md (hunk 1 above)
Comment (hdevalence) [reply site/formulas/decoding.md:10]: Yeah, the code is updated in a different PR, it swapped naming of u_1 and u_2

**Comment by hdevalence** [comment]
Merging this since the part that's out of sync is internal to the website notes and there's a code PR to go with

**Pull request merged**